config/listCheckerConfig.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"istio.io/mixer/adapters"
+	"errors"
+)
+
+type ListCheckerConfig struct {
+	AdapterConfig *adapters.AdapterConfig
```

Comment:
As discussed earlier, we can update the code in such a way i.e.
----------
type ListCheckerEvaluationConfig struct {
// Only config for evaluation purposes
}

// Given a config and facts, evaluate the symbol value to use.
func EvaluateListcheckerConfig(labels map[string]string, ListCheckerEvaluationConfig) (string, error)

In the apiHandler:
adapterConfig, evaluationConfig, err := configManager.getConfig(peerID, serviceName)
adapter, err := adapterManager.getAdapter(peerID, serviceName, adapterConfig)

symbolValue := EvaluateListCheckerConfig(labelValues, evaluationConfig)

adapter.CheckList(symbolValue)

----------
However, I'd expect the machinery inside the "EvaluateListCheckerConfig" to be amenable to caching, especially if the config language is complicated and has expression support (which seems to be the case). As such, my inclination was to make the ConfigManager a caching component, similar to AdapterManager, and hold onto evaluators. This makes them effectively interfaces, similar to the current design here. (Except my choice of names are a bit off. ListCheckerConfig should probably be ListCheckerConfigEvaluator and instead of having an attached adapter config, it would return the adapter config as one of the result values. (Still adjusting to Go model of doing things).

If you buy into this view (i.e. cached evaluators) then I can update the code to be a bit more coherent?